internal/beater/beater.go

```diff
@@ -537,6 +537,50 @@ func (s *Runner) Run(ctx context.Context) error {
 	return errors.Join(result, closeErr)
 }
 
+// newInstrumentation is a thin wrapper around libbeat instrumentation that
+// sets missing tracer configuration from elastic-agent.
+func newInstrumentation(rawConfig *agentconfig.C) (instrumentation.Instrumentation, error) {
```

Comment:
A unit and/or system test would be great to have.

Reply:
The test is added. Unfortunately, it's not possible to test this change with a unit test. So I added a small functional test instead that covers the added functionality.